### PR TITLE
[expo-updates] Fix metro asset call in embedded manifest creation step

### LINKED ISSUE #26195: [SDK 50] expo-updates + src dir breaks assets
### Minimal reproducible example

https://github.com/alexandernanberg/expo-rmissing-image-epro

### Summary

When using `expo-updates` in combination with the `src` directory all images (probably other assets too assets?) disappear in production builds. 

Steps to reproduce
1. Clone repro repo
2. Setup expo-updates with a project you can test with
3. Start dev and check that the image renders
4. Create a production build an install it, notice that now the image isn't showing anymore

<details>
<summary>Screenshots</summary>

#### Dev
Simulator Screenshot - iPhone 15 Pro - 2024-01-04 at 08 31 08

#### Prod
Simulator Screenshot - iPhone 15 Pro - 2024-01-04 at 08 28 44

</details>

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 14.2.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.9.0 - ~/.nvm/versions/node/v20.9.0/bin/node
      npm: 10.1.0 - ~/.nvm/versions/node/v20.9.0/bin/npm
    Managers:
      CocoaPods: 1.14.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.2, iOS 17.2, macOS 14.2, tvOS 17.2, watchOS 10.2
    IDEs:
      Android Studio: 2022.3 AI-223.8836.35.2231.10811636
      Xcode: 15.1/15C65 - /usr/bin/xcodebuild
    npmPackages:
      expo: 50.0.0-preview.7 => 50.0.0-preview.7 
      react: 18.2.0 => 18.2.0 
      react-native: 0.73.1 => 0.73.1 
    npmGlobalPackages:
      eas-cli: 5.9.3
    Expo Workflow: bare
```

### PR DESCRIPTION
**Pull request by wschurman**
# Why

The metro asset collection step in `@expo/cli` has evolved and diverged from the way that the expo-updates asset collection step was written. This causes nonstandard metro configs to not be respected by expo-updates, and assets to be missing.

We didn't notice this for a while because until https://github.com/expo/expo/pull/25613 landed a lot of common assets were present in the CDN and would resolve and be downloaded instead of getting the embedded ones. (they'd still present on screen, just erroneously download).

Closes #26195.
Closes ENG-11025.

# How

This fix is done in two commits. The first commit is sufficient to fix the issue (it copy/pastes the code from CLI into createManifest but still leaves the root issue open as it could diverge again in the future). The second commit tries to future-proof this by factoring out the functions into common functions used by both the CLI export and the expo-updates export.

# Test Plan

copy-paste the built JS for `createManifest` and `exportEmbedAsync` into a sample app that this repro's in. the repro is this comment: https://github.com/expo/expo/issues/26195#issuecomment-1880199885

Inspect the output of createManifest and see the image in a release build.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-updates/scripts/createManifest.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
-const { loadMetroConfigAsync } = require('@expo/cli/build/src/start/server/metro/instantiateMetro');
```
Comment (wschurman) [inline packages/expo-updates/scripts/createManifest.js:1]: `expo-updates` doesn't have an explicit dependency on `@expo/cli`. should it?

Comment (douglowder) [reply packages/expo-updates/scripts/createManifest.js:1]: It should derive its `@expo/cli` version from the `expo` version, I would think

**Review by douglowder — APPROVED** [review]

File: packages/expo-updates/scripts/createManifest.js (hunk 1 above)
Comment (wschurman) [reply packages/expo-updates/scripts/createManifest.js:1]: Does `expo` depend on `@expo/cli`? (explicitly, in `package.json`)

Comment (douglowder) [reply packages/expo-updates/scripts/createManifest.js:1]: Yes indeed, see `packages/expo/package.json`

**Review by EvanBacon — APPROVED** [review]

**Pull request merged**